python/equistore-operations/equistore/operations/manipulate_metadata.py

```diff
@@ -0,0 +1,335 @@
+"""
+Manipulating Metadata
+=====================
+
+Functions for manipulating metadata/:py:class:`equistore.Labels` of a
+:py:class:`equistore.TensorMap`.
+
+.. autofunction:: equistore.append_metadata
+
+.. autofunction:: equistore.remove_metadata
+
+.. autofunction:: equistore.rename_metadata
+"""
+from typing import List, Sequence, Union
+
+import numpy as np
+
+from equistore.core import Labels, TensorBlock, TensorMap
+
+
+def _labels_append(labels: Labels, name, values: Union[int, np.ndarray]) -> Labels:
+    """Append a new column with the given name and values to labels
+
+    Maybe implement as a Labels method.
+    """
+
+    if type(values) == int:
+        values = values * np.ones(len(labels), dtype=int).reshape(-1, 1)
+    elif type(values) == np.ndarray and len(values.shape) != 2:
+        raise ValueError("`values` must be a 2D array")
+
+    if len(values) != len(labels):
+        raise ValueError(
+            f"length of to be appended values ({len(values)}) does not match with "
+            f"those in labels ({len(labels)})"
+        )
+
+    new_values = np.hstack([labels.values, np.reshape(values, (-1, 1))])
+    new_names = labels.names + [name]
+
+    return Labels(names=new_names, values=new_values)
+
+
+def _remove_from_list(a: List, b: List):
+    """Remove all items in ``b`` from ``a``"""
+    for item in b:
+        if item not in a:
+            raise ValueError(f"'{item}' not in list")
+        a.remove(item)
+
+
+def append_metadata(
+    tensor: TensorMap, name: str, values: Union[float, np.ndarray], axis: str
```

Comment:
I currently only allow for only one new label name. We can extend this to allow also a list of names. However, this will makes verifying the values more complex...

Reply:
Yes I think a single name is fine for now